railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -19,15 +19,29 @@ def rakefile
     def app
       if mountable?
         if api?
-          directory "app", exclude_pattern: %r{app/(views|helpers)}
+          excluded_directories = %w{ views helpers }
+          excluded_directories << "mailers" if options[:skip_action_mailer]
+          excluded_directories << "jobs" if options[:skip_active_job]
+
+          directory "app", exclude_pattern: %r{app/(#{excluded_directories.join("|")})}
         else
-          directory "app"
+          excluded_directories = []
+          excluded_directories << "mailers" if options[:skip_action_mailer]
+          excluded_directories << "jobs" if options[:skip_active_job]
+
+          if excluded_directories.empty?
+            directory "app"
+          else
+            directory "app", exclude_pattern: %r{app/(#{excluded_directories.join("|")})}
+          end
```

Comment:
Using `exclude_pattern` here seems to be causing a lot of complexity, and dynamically generating a regex makes the logic quite hard to follow. What if we did the same thing as the app generator and [removed](https://github.com/rails/rails/blob/b39ea7cc3662345ad8efbe496db5ae594449f85f/railties/lib/rails/generators/rails/app/app_generator.rb#L504) the [directories](https://github.com/rails/rails/blob/b39ea7cc3662345ad8efbe496db5ae594449f85f/railties/lib/rails/generators/rails/app/app_generator.rb#L496) after copying them instead? The diff would be much simpler:

```patch
diff --git a/railties/lib/rails/generators/rails/plugin/plugin_generator.rb b/railties/lib/rails/generators/rails/plugin/plugin_generator.rb
index d59296a64e..7db6d9b63f 100644
--- a/railties/lib/rails/generators/rails/plugin/plugin_generator.rb
+++ b/railties/lib/rails/generators/rails/plugin/plugin_generator.rb
@@ -25,10 +25,14 @@ def app
           directory "app"
           empty_directory_with_keep_file "app/assets/images/#{namespaced_name}"
         end
+
+        remove_dir "app/mailers" if options[:skip_action_mailer]
+        remove_dir "app/jobs" if options[:skip_active_job]
       elsif full?
         empty_directory_with_keep_file "app/models"
         empty_directory_with_keep_file "app/controllers"
-        empty_directory_with_keep_file "app/mailers"
+        empty_directory_with_keep_file "app/mailers" unless options[:skip_action_mailer]
+        empty_directory_with_keep_file "app/jobs" unless options[:skip_active_job]

         unless api?
           empty_directory_with_keep_file "app/assets/images/#{namespaced_name}"
```

Reply:
@eugeneius PR updated, please take another look, thanks.